models/account.go

```diff
@@ -2,20 +2,41 @@
 // of the accounts service.
 package models
 
-import "github.com/google/uuid"
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
 
 type Account struct {
-	ID uuid.UUID `json:"id"`
+	ID    uuid.UUID `json:"id" bson:"_id,omitempty"`
+	Email string    `json:"email" bson:"email,omitempty"`
+	Name  string    `json:"name" bson:"name,omitempty"`
+	Hash  *[]byte   `json:"hash" bson:"hash,omitempty"`
+}
+
+type AccountPayload struct {
+	Name  *string `json:"name" bson:"name,omitempty"`
+	Email *string `json:"email" bson:"email,omitempty"`
+	Hash  *[]byte `json:"hash" bson:"hash,omitempty"`
 }
 
-type AccountQuery struct {
-	ID    uuid.UUID `json:"id"`
-	Email string    `json:"email"`
+type OneAccountFilter struct {
+	ID    uuid.UUID `json:"id" bson:"_id,omitempty"`
+	Email string    `json:"email" bson:"email,omitempty"`
 }
 
+type ManyAccountFilter struct{}
+
 // AccountsRepository is safe for use in multiple goroutines.
 type AccountsRepository interface {
-	Get(filter *AccountQuery) (*Account, error)
+	Create(ctx context.Context, filter *AccountPayload) error
+
+	Get(ctx context.Context, filter *OneAccountFilter) (*Account, error)
+
+	Delete(ctx context.Context, filter *OneAccountFilter) error
+
+	Update(ctx context.Context, filter *OneAccountFilter, account *AccountPayload) error
 
-	Delete(filter *AccountQuery) error
+	List(ctx context.Context) (*[]Account, error)
```

Comment:
List could take some filter parameters and some pagination helpers (like limit, offset, etc). We'll do that next sprint